util/stream.go

```diff
@@ -135,61 +121,107 @@ func (m *MessageStream) outbound() {
 	}
 }
 
+type smallMessageError struct {
+	length int
+}
+
+func (e smallMessageError) Error() string {
+	return fmt.Sprintf("invalid message with length %d is received", e.length)
+}
+
+func newSmallMessageError(length int) *smallMessageError {
+	return &smallMessageError{
+		length: length,
+	}
+}
+
+func getMessageLength(reader *bufio.Reader) (int, error) {
```

Comment:
This function would benefit from a detailed comment

Reply:
Added.